### ISSUE TITLE
DeterministicKey::toString skips creationTimeSeconds when parent key has creationTimeSeconds > 0

### ISSUE THREAD

**Issue by msgilligan** [bitcoinj/bitcoinj#1899]
`.toString()` in `DeterministicKey` uses the following code to skip printing `creationTimeSeconds` when it is zero:

```
        if (creationTimeSeconds > 0)
            helper.add("creationTimeSeconds", creationTimeSeconds);
```

However, the getter will return the creationTimeSeconds from the `parent` key:

```
    public long getCreationTimeSeconds() {
        if (parent != null)
            return parent.getCreationTimeSeconds();
        else
            return super.getCreationTimeSeconds();
```

`toString` should probably call the getter instead of accessing the field.

**Comment by msgilligan** [comment]
@schildbach If you agree, assign this to me and I'll make a PR.

**Comment by schildbach** [comment]
> toString should probably call the getter instead of accessing the field.

We could do this, but then I think we should add an info wether the value is inherited or not.

**Comment by msgilligan** [comment]
I think we should do it. I was confused when debugging because the key didn't have a creation time when I expected it to have one. I'll make a PR.

**Comment by msgilligan** [comment]
See PR #1903

**Comment by schildbach** [comment]
Fixed via b59add47a7aba1f3bec8356b567bf0df4b036f81.
